#!/usr/bin/env python
# field of view configuration

PACKAGE="shape_finder"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("MeanK", int_t, 1, "MeanK", 50, 1, 500)
gen.add("StddevMulThresh", double_t, 1, "StddevMulThresh", 0.5, 0.0, 10.0)
gen.add("overlap_threshold", double_t, 1, "Overlap_threshold", 0.6, 0.001, 1)
gen.add("big_plane_size", int_t, 1, "big_plane_size", 600, 10, 10000)
gen.add("min_cloud_size", int_t, 1, "min_cloud_size", 50, 10, 10000)
gen.add("hv_tolerance", double_t, 1, "hv_tolerance", 15.0, 0.0, 90.0)
gen.add("DistanceThresholdPlane0", double_t, 1, "DistanceThresholdPlane0", 0.01, 0.0, 5.0)
gen.add("NormalDistanceWeightPlane", double_t, 1, "NormalDistanceWeightPlane", 0.035, 0.0, 5.0)
gen.add("MaxIterationsPlane", int_t, 1, "MaxIterationsPlane", 50, 1, 100000)
gen.add("DistanceThresholdPlane", double_t, 1, "DistanceThresholdPlane", 0.01, 0.0, 5.0)
gen.add("NormalDistanceWeightCyilnder", double_t, 1, "NormalDistanceWeightCyilnder", 0.1, 0.0, 5.0)
gen.add("MaxIterationsCylinder", int_t, 1, "MaxIterationsCylinder", 750, 1, 100000)
gen.add("DistanceThresholdCylinder", double_t, 1, "DistanceThresholdCylinder", 1.0, 0.0, 5.0)
gen.add("RadiusLimitsMinCylinder", double_t, 1, "RadiusLimitsMinCylinder", 0.0, 0.0, 1.0)
gen.add("RadiusLimitsMaxCylinder", double_t, 1, "RadiusLimitsMaxCylinder", 0.3, 0.0, 1.0)


exit(gen.generate(PACKAGE, "shape_finder", "ShapeFinderNodelet"))
